src/playlist/PlaylistSong.cxx

```diff
@@ -54,18 +56,22 @@ merge_song_metadata(DetachedSong &add, const DetachedSong &base) noexcept
 }
 
 static bool
-playlist_check_load_song(DetachedSong &song, const SongLoader &loader) noexcept
-try {
-	DetachedSong tmp = loader.LoadSong(song.GetURI());
-
-	song.SetURI(tmp.GetURI());
-	if (!song.HasRealURI() && tmp.HasRealURI())
-		song.SetRealURI(tmp.GetRealURI());
-
-	merge_song_metadata(song, tmp);
-	return true;
-} catch (...) {
-	return false;
+playlist_check_load_song(DetachedSong &song, const SongLoader &loader) noexcept {
+	try {
+		DetachedSong tmp = loader.LoadSong(song.GetURI());
+
+		song.SetURI(tmp.GetURI());
+		if (!song.HasRealURI() && tmp.HasRealURI())
+			song.SetRealURI(tmp.GetRealURI());
+
+		merge_song_metadata(song, tmp);
+		return true;
+	} catch (const std::exception &ex) {
```

Comment:
If the exception is not derived from `std::exception`, then MPD crashes.

Reply:
Ah! Is it the case that exceptions _not_ derived from `std::exception` may be thrown in MPD? If so, I'll update.